src/Hintman/Server/Auth.hs

```diff
@@ -21,11 +28,37 @@ type instance AuthServerData (AuthProtect "GitHub") = ()
 type HintmanAuthContextHandlers = '[AuthHandler Request ()]
 type HintmanAuthContext = Context HintmanAuthContextHandlers
 
+verifySignature :: Request -> Handler ()
+verifySignature req = do
+    body <- liftIO $ requestBody req
+    verifySignature' body header
+  where
+    header :: Maybe ByteString
+    header = lookup "x-hub-signature" (requestHeaders req)
+
+verifySignature' :: ByteString -> Maybe ByteString -> Handler ()
+verifySignature' _ Nothing = throwError $ err401 { errBody = "Empty signature." }
+verifySignature' payload (Just signature) = do
+    let (_, digest) = breakSubstring signature "="
+    let theirDigest = digestFromByteString digest
+    key <- liftIO $ pack <$> getEnv "GITHUB_WEBHOOK_SECRET"
```

Comment:
I would love to see PR with my JWT functions merged, so we can have various functions to ask parts of the configuration from environment

Reply:
@ChShersh just pushed a rebase with the new types! :)